frontend/exporter/src/types/copied.rs

```diff
@@ -1617,12 +1619,17 @@ pub enum Ty {
         rustc_middle::ty::TyKind::Adt(adt_def, substs) => {
             let def_id = adt_def.did().sinto(state);
             let generic_args: Vec<GenericArg> = substs.sinto(state);
-            Ty::Adt { def_id, generic_args }
+            let param_env = state.base().tcx.param_env(state.owner_id());
+            let trait_refs = solve_item_traits(state, param_env, adt_def.did(), substs, None);
```

Comment:
Just as a not here, I filed https://github.com/hacspec/hax/issues/637
That's what we discussed the other day, but that will be for another PR.